File: crates/ffi/src/platforms/windows/AdGuard.FilterListManagerProtobuf/AdGuard.FilterListManagerProtobuf/RustInterface/RustResponseType.cs
namespace AdGuard.FilterListManagerProtobuf.RustInterface
{
    /// <summary>
    /// Discriminant for [`RustResponse`] result_data value
    /// </summary>
    enum RustResponseType
    {
        /// <summary>
        /// Contains u8 pointer with size
        /// </summary>
        RustBuffer,
        
        /// <summary>
        /// Contains [`FLMHandle`]
        /// </summary>
        FLMHandlePointer
    }
}

File: crates/ffi/src/platforms/kotlin-multiplatform/lib/src/commonMain/kotlin/com/adguard/flm/support/RustResponseType.kt
package com.adguard.flm.support

/**
 * Copy for native enum from [RustResponseType](../../../../../../../../../flm_native_interface.h).
 * These 2 enums must be synchronized.
 */
enum class RustResponseType {
    /**
     * Contains a byte buffer with data (e.g., protobuf serialized data)
     */
    RustBuffer,

    /**
     * Contains a handle to a Filter List Manager instance
     */
    FLMHandlePointer
}
